Libplanet/Blockchain/Renderers/NonblockRenderer.cs

```diff
@@ -0,0 +1,195 @@
+                FullMode = boundedChannelFullMode,
+                SingleReader = true,
+                SingleWriter = true,
+            });
+            _writer = _channel.Writer;
+            _reader = _channel.Reader;
+            _fullFallback = fullFallback;
+            _worker = new Thread(async () =>
+            {
+                while (await _reader.WaitToReadAsync())
+                {
+                    while (_reader.TryRead(out System.Action? action))
+                    {
+                        action?.Invoke();
+                    }
+                }
+            })
+            {
+                IsBackground = true,
+            };
+        }
+
+        /// <summary>
+        /// Customizes behavior when the internal event queue is full so that no more event
+        /// can be added.
+        /// </summary>
+        /// <param name="droppedEvent">The render event failed to be queued.</param>
+        public delegate void FullFallback(System.Action droppedEvent);
+
+        /// <summary>
+        /// Specifies the behavior when the internal event queue is full so that no more event
+        /// can be added.
+        /// </summary>
+        public enum FullMode
+        {
+            /// <summary>
+            /// Drops the oldest event when the queue is full.
+            /// </summary>
+            DropOldest,
+
+            /// <summary>
+            /// Drops the newest event when the queue is full.
+            /// </summary>
+            DropNewest,
+        }
+
+        /// <summary>
+        /// The inner renderer which has the <em>actual</em> implementations and receives events.
+        /// </summary>
+        public IRenderer<T> Renderer { get; }
+
+        /// <inheritdoc cref="IDisposable.Dispose()"/>
+        public void Dispose()
+        {
+            _channel.Writer.Complete();
+            if (_worker.IsAlive)
+            {
+                _worker.Join();
+            }
+        }
+
+        /// <inheritdoc cref="IRenderer{T}.RenderBlock(Block{T}, Block{T})"/>
+        public void RenderBlock(Block<T> oldTip, Block<T> newTip) =>
+            Queue(() => Renderer.RenderBlock(oldTip, newTip));
+
+        /// <inheritdoc cref="IRenderer{T}.RenderReorg(Block{T}, Block{T}, Block{T})"/>
+        public void RenderReorg(Block<T> oldTip, Block<T> newTip, Block<T> branchpoint) =>
+            Queue(() => Renderer.RenderReorg(oldTip, newTip, branchpoint));
+
+        /// <inheritdoc cref="IRenderer{T}.RenderReorgEnd(Block{T}, Block{T}, Block{T})"/>
+        public void RenderReorgEnd(Block<T> oldTip, Block<T> newTip, Block<T> branchpoint) =>
+            Queue(() => Renderer.RenderReorgEnd(oldTip, newTip, branchpoint));
+
+        /// <summary>
+        /// Queues the callback to be executed in the worker thread.
+        /// </summary>
+        /// <param name="action">The callback to be executed in the worker thread.</param>
+        protected void Queue(System.Action action)
```

Comment:
It also looks similar to[`TaskScheduler.QueueTask`](https://docs.microsoft.com/ko-kr/dotnet/api/system.threading.tasks.taskscheduler.queuetask?view=net-5.0). It would have been nice to be able to use an implementation like `SingleThreadedTaskScheduler`, but there is no implementation in advance. 😢